math/binary/sqrt.go

```diff
@@ -0,0 +1,26 @@
+// sqrt.go
+// description: Square root calculation
+// details:
+// Calculating the square root using binary operations and a magic number 0x5f3759df [See more](https://en.wikipedia.org/wiki/Fast_inverse_square_root)
+// author(s) [red_byte](https://github.com/i-redbyte)
+// see sqrt_test.go
+
+package binary
+
+import (
+	"math"
+)
+
+const threeHalves = 1.5
+
+func Sqrt(number float32) float64 {
```

Comment:
I'm curious to see which is faster, the assembly sqrt function vs this one. Assembly one is usually faster in C/C++ but sometime the speed for assembly doesn't translate in Go. I saw that you added the benchmark comparing it to the `math` sqrt - will check it out when I get time.

Reply:
in my PC(MacBook Pro (16-inch, 2019) Intel Core i9 2,3 DDR4 32 Gb):

BenchmarkSquareRootCalculation-16    	1000000000	         0.2363 ns/op
BenchmarkMathSqrt-16    	                         1000000000	         0.2466 ns/op

Reply:
Very interesting, I feel like the conversion [here](https://cs.opensource.google/go/go/+/refs/tags/go1.17.8:src/math/sqrt.go;drc=1d20a362d0ca4898d77865e314ef6f73582daef0;l=92) to assembly instruction that happens in the `math.Sqrt` function is taking the most time but I think I will try to do some more profiling to figure out why. Will be busy this whole coming week so probably won't get it done anytime soon 😓